codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java

```diff
@@ -176,44 +179,52 @@ private static MqttFixedHeader decodeFixedHeader(ByteBuf buffer) {
      * @param mqttFixedHeader MqttFixedHeader of the same message
      * @return the variable header
      */
-    private static Result<?> decodeVariableHeader(ByteBuf buffer, MqttFixedHeader mqttFixedHeader) {
+    private Result<?> decodeVariableHeader(ChannelHandlerContext ctx, ByteBuf buffer, MqttFixedHeader mqttFixedHeader) {
```

Comment:
can't this be static now ?

Reply:
Unfortunately no - `decodePubReplyMessage`, `decodePublishVariableHeader` and `decodeReasonCodeAndPropertiesVariableHeader` rely on `bytesRemainingInVariablePart`, because reason code and properties may be skipped in case of successful message.